test/ng/directive/ngOptionsSpec.js

```diff
@@ -2217,6 +2217,21 @@ describe('ngOptions', function() {
       dealoc(element);
     }));
 
+    // See issue #13331
+    it('should set [selected] to false when a genuine option is selected', function() {
+      scope.options = ['a'];
+      scope.model = '';
+      compile(
+          '<select ng-model="model" ng-options="opt for opt in options">' +
+            '<option id="empty_option" value="">Empty Option</option>' +
+          '</select>');
+      var opt = element.find("option")[0];
+      expect(opt.getAttribute("selected")).toBeTruthy();
+      scope.model = 'a';
+      scope.$digest();
+      expect(opt.getAttribute("selected")).toBeFalsy();
```

Comment:
Wouldn't it be better to check with `hasAttribute()` ?
Even an empty (falsy) attribute value (e.g. `selected=""`) might mean the option is selected.

I would also check the property for good measure (here and above).